src/result.js

```diff
@@ -127,29 +127,25 @@ function missingTag(tag) {
 
 function insertIntoIndexHTML(html, head, body) {
   if (!html) { return Promise.resolve(html); }
+  let isBodyReplaced = false;
+  let isHeadReplaced = false;
 
-  if (body) {
-    let isBodyReplaced = false;
-    html = html.replace("<!-- EMBER_CLI_FASTBOOT_BODY -->", function() {
+  html = html.replace(/<\!-- EMBER_CLI_FASTBOOT_(HEAD|BODY) -->/g, function(match, p1) {
```

Comment:
My two cents:

I really find the changes convoluted and the `head` and `body` fork is at two places instead of one - making readability hard. I am not convinced this is improving readability if that is what the motivation of the changes was.